### ISSUE TITLE
CI authentication has problems

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#12029]
This morning a scheduled website test failed and shows the following error in the log:

```
Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --force --progress git://github.com/cgeo/cgeo.git +refs/heads/*:refs/remotes/origin/* --prune" returned status code 128:
stdout: 
stderr: fatal: remote error: 
  The unauthenticated git protocol on port 9418 is no longer supported.
Please see https://github.blog/2021-09-01-improving-git-protocol-security-github/ for more information.
```
https://github.blog/2021-09-01-improving-git-protocol-security-github/ 

Job log: See https://ci.cgeo.org/job/website%20tests%20(release)/10209/display/redirect

Strange enough the next scheduled build afterwards was OK again.
Did someone upgrade the keys or was this just an early warning as a reminder, that we need to change something?

**Comment by Lineflyer** [comment]
<html>
<body>


November 2, 2021 | First brownout; RSA with SHA-1 cutoff.All user RSA keys with valid_after dates after this point will need to use SHA-2 signatures during the brownout periods and after the change becomes permanent. We’ll also run several short brownouts on this date. During a brownout, the MACs, ciphers, and protocol we’re removing will be temporarily disabled.
-- | --

</body>
</html>

That seems to be the issue here...brownout phase

**Comment by Lineflyer** [comment]
Should be solved. What I changed:
- Created a SSH key pair 
- Stored private key as SSH key for user `cgeo-ci-bot` on Github
- Stored public key in Jenkins credential manager (global)
- Either created a copy of existing relevant jobs or changed existing jobs to use the SSH login instead of unauthenticated fetching
- Cleaned up a bit by deleted old jobs/projects on CI, which are well outdated

By now I hope I changed all relevant jobs for CI (PR + commit builds) and nightly and release builds for both cgeo and WhereYouGo.

I will close here, but follow up on this issue if you notice failing jobs with related errors or missing builds/nightlies,

**Comment by Lineflyer** [comment]
Just a "heads up":

I changed the commit and PR build timeout configuration on CI as we happen to have stuck builds (emulator dead) from time to time. It was previously set to an absolute limit of 30 minutes.

I changed it to an elastic limit of 150% of the mean time of the last 3 builds and a maximum of 20 minutes.
That should free the build processor a bit earlier than today. 

If it turns out to terminate normal builds in progress, let me know to fine tune the settings.
